домашняя работа спринт 11

KozlovskiyD opened the pull request:
Toast all
Домашная работа часть 3.5
Домашная работа спринт 9.1
Домашняя работа спринт 10.1
домашняя работа спринт 11

Dmitr80 committed:
Init commit
Toast all
Домашная работа часть 3.5
Домашная работа спринт 9.1
Домашняя работа спринт 10.1
домашняя работа спринт 11

SeveNChaK commented on app/src/main/java/com/practicum/playlistmaker3/SearchActivity.kt:
```diff
@@ -37,29 +66,94 @@ class SearchActivity : AppCompatActivity() {
+                        modelTrack.clear()
+                        if (response.body()?.results?.isNotEmpty() == true) {
+                            modelTrack.addAll(response.body()?.results!!)
+                            trackListAdapter.notifyDataSetChanged()
```
⏫ Рекомендую сделать в адаптере метод
```
fun setTracks(newTracks: List<Track>?) {
        tracks.clear()
        if (!newTracks.isNullOrEmpty()) {
            tracks.addAll(newTracks)
        }
        notifyDataSetChanged()
}
```
А список в классе адаптера объявить вот так `private val tracks = mutableListOf<Track>()`. Тогда не надо будет во всех местах писать по несколько строк, достаточно будет обратиться к методу адаптера и передать нужные данные: `adapter.setTracks(новыйСписок)` - если хотим установить новый список, `adapter.setTracks(null)` - если хотим почистить список треков.

И переменную списка из активити можно удалить в таком случае.

SeveNChaK commented on app/src/main/java/com/practicum/playlistmaker3/TrackListHolder.kt:
```diff
@@ -1,30 +1,39 @@
             .load(model.artworkUrl100)
             .placeholder(R.drawable.ic_launcher_foreground)
             .centerCrop()
             .transform(RoundedCorners(10))
```
⚠️ Здесь радиус задается в пикселях. Поэтому значение из макета (которое мы берем как значение в `dp`) необходимо преобразовать в пиксели.
Удобнее всего будет вынести это значение в ресурсы, а потом достать его здесь: `itemView.resources.getDimensionPixelSize(нужный ресурс)`

Значение в макете было 2

SeveNChaK commented on app/src/main/java/com/practicum/playlistmaker3/TrackListHolder.kt:
```diff
@@ -1,30 +1,39 @@
         Glide
             .with(itemView)
             .load(model.artworkUrl100)
             .placeholder(R.drawable.ic_launcher_foreground)
```
⚠️ Не тот плейсхолдер
<img width="137" alt="image" src="https://user-images.githubusercontent.com/22057568/234927285-9c71f093-d580-47cf-8b4e-96d802d79f1b.png">

В макете
<img width="181" alt="image" src="https://user-images.githubusercontent.com/22057568/234927215-98e37330-6572-467d-adad-6ad9d61e495f.png">

SeveNChaK commented on app/build.gradle:
```diff
@@ -40,9 +40,16 @@ dependencies {
     implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
     implementation 'com.github.bumptech.glide:glide:4.15.1'
     implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
```
⏫ Дублируется, можно убрать

SeveNChaK commented on app/build.gradle:
```diff
@@ -40,9 +40,16 @@ dependencies {
     implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
     implementation 'com.github.bumptech.glide:glide:4.15.1'
     implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
+    implementation 'com.android.volley:volley:1.2.1'
```
⏫ Не нужный импорт, можно убрать

SeveNChaK commented on app/build.gradle:
```diff
@@ -40,9 +40,16 @@ dependencies {
     implementation 'com.github.bumptech.glide:glide:4.15.1'
     implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
+    implementation 'com.android.volley:volley:1.2.1'
+    implementation 'androidx.compose.ui:ui-graphics:1.1.1'
```
⏫ Не нужный импорт, можно убрать

SeveNChaK commented on app/build.gradle:
```diff
@@ -40,9 +40,16 @@ dependencies {
     implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
+    implementation 'com.android.volley:volley:1.2.1'
+    implementation 'androidx.compose.ui:ui-graphics:1.1.1'
+    implementation 'com.github.bumptech.glide:glide:4.14.2'
```
⏫ дублируется, можно убрать

SeveNChaK commented on app/build.gradle:
```diff
@@ -40,9 +40,16 @@ dependencies {
+    implementation 'com.android.volley:volley:1.2.1'
+    implementation 'androidx.compose.ui:ui-graphics:1.1.1'
+    implementation 'com.github.bumptech.glide:glide:4.14.2'
     implementation 'androidx.navigation:navigation-fragment-ktx:2.5.3'
```
⏫ Не нужный импорт, можно убрать

SeveNChaK commented on app/build.gradle:
```diff
@@ -40,9 +40,16 @@ dependencies {
+    implementation 'androidx.compose.ui:ui-graphics:1.1.1'
+    implementation 'com.github.bumptech.glide:glide:4.14.2'
     implementation 'androidx.navigation:navigation-fragment-ktx:2.5.3'
+    implementation 'androidx.contentpager:contentpager:1.0.0'
```
⏫ Не нужный импорт, можно убрать

SeveNChaK commented on app/build.gradle:
```diff
@@ -40,9 +40,16 @@ dependencies {
+    implementation 'androidx.contentpager:contentpager:1.0.0'
+    implementation 'com.squareup.retrofit2:retrofit:2.3.0'
+    implementation 'com.squareup.retrofit2:converter-gson:2.3.0'
+    implementation 'com.squareup.retrofit2:adapter-rxjava2:2.3.0'
```
⏫ Не нужный импорт, можно убрать

SeveNChaK commented on app/build.gradle:
```diff
@@ -40,9 +40,16 @@ dependencies {
     androidTestImplementation 'androidx.test.ext:junit:1.1.5'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.5.1'
-    annotationProcessor 'com.github.bumptech.glide:compiler:4.15.1'
+    annotationProcessor 'com.github.bumptech.glide:compiler:4.14.2'
```
⏫ А почему понизил версию? Кажется, что можно вернуть 4.15.1

SeveNChaK reviewed (commented):
⏫ не однотонный фон на экране поиска
<img width="406" alt="image" src="https://user-images.githubusercontent.com/22057568/234928957-97baf3d1-2b9b-4379-8a55-8d0a77eb1e2b.png">

Dmitr80 committed:
Init commit
Toast all
Домашная работа часть 3.5
Домашная работа спринт 9.1
Домашняя работа спринт 10.1
домашняя работа спринт 11.1
